capella2polarion/connectors/polarion_worker.py

```diff
@@ -425,36 +461,104 @@ def compare_and_update_work_items(
             if uuid in self.polarion_data_repo and data.work_item is not None:
                 self.compare_and_update_work_item(data)
 
-    def post_documents(self, documents: list[polarion_api.Document]):
+    def create_documents(
+        self,
+        document_datas: list[data_models.DocumentData],
+        document_project: str | None = None,
+    ):
         """Create new documents."""
-        self.client.project_client.documents.create(documents)
+        client = self._get_client(document_project)
+        documents, _ = self._process_document_datas(client, document_datas)
+
+        client.documents.create(documents)
 
-    def update_documents(self, documents: list[polarion_api.Document]):
+    def update_documents(
+        self,
+        document_datas: list[data_models.DocumentData],
+        document_project: str | None = None,
+    ):
         """Update existing documents."""
-        self.client.project_client.documents.update(documents)
+        client = self._get_client(document_project)
+        documents, headings = self._process_document_datas(
+            client, document_datas
+        )
+
+        client.work_items.update(headings)
+        client.documents.update(documents)
+
+    def _process_document_datas(
+        self,
+        client: polarion_api.ProjectClient,
+        document_datas: list[data_models.DocumentData],
+    ):
+        documents = []
+        headings = []
```

Comment:
I don't think we are using mypy with the pre-commit hooks anymore. My mypy would complain here because of missing type-annotations.
```suggestion
        documents: list[polarion_api.Document] = []
        headings: list[polarion_api.WorkItem] = []
```